### ISSUE TITLE
run on ios react-native 0.60 build failed with error

### ISSUE THREAD

**Issue by Zbish** [negativetwelve/react-native-ux-cam#87]
```
Ld /Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/Venn.app/Venn normal x86_64 (in target: Venn)
    cd /Users/omridoron/dev/venn/mobile-app-v2/ios
    export IPHONEOS_DEPLOYMENT_TARGET=9.0
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator12.1.sdk -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/DoubleConversion -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/Folly -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/Google-Maps-iOS-Utils -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/JKBigInteger2 -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/RNAWSCognito -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/RNCAsyncStorage -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/RNFastImage -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/RNGestureHandler -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/RNKeychain -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/RNLocalize -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/RNSVG -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/RNUxcam -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-Core -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-DevSupport -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-RCTActionSheet -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-RCTAnimation -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-RCTBlob -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-RCTImage -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-RCTLinking -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-RCTNetwork -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-RCTSettings -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-RCTText -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-RCTVibration -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-RCTWebSocket -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-cxxreact -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-fishhook -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-jsi -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-jsiexecutor -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/React-jsinspector -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/SDWebImage -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/SDWebImageWebPCoder -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/glog -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/libwebp -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/react-native-google-maps -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/react-native-maps -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/react-native-splash-screen -L/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/yoga -F/Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator -F/Users/omridoron/dev/venn/mobile-app-v2/ios/Pods/GoogleMaps/Base/Frameworks -F/Users/omridoron/dev/venn/mobile-app-v2/ios/Pods/GoogleMaps/Maps/Frameworks -filelist /Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Intermediates.noindex/Venn.build/Debug-iphonesimulator/Venn.build/Objects-normal/x86_64/Venn.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -mios-simulator-version-min=9.0 -Xlinker -object_path_lto -Xlinker /Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Intermediates.noindex/Venn.build/Debug-iphonesimulator/Venn.build/Objects-normal/x86_64/Venn_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -ObjC -lDoubleConversion -lFolly -lGoogle-Maps-iOS-Utils -lJKBigInteger2 -lRNAWSCognito -lRNCAsyncStorage -lRNFastImage -lRNGestureHandler -lRNKeychain -lRNLocalize -lRNSVG -lRNUxcam -lReact-Core -lReact-DevSupport -lReact-RCTActionSheet -lReact-RCTAnimation -lReact-RCTBlob -lReact-RCTImage -lReact-RCTLinking -lReact-RCTNetwork -lReact-RCTSettings -lReact-RCTText -lReact-RCTVibration -lReact-RCTWebSocket -lReact-cxxreact -lReact-fishhook -lReact-jsi -lReact-jsiexecutor -lReact-jsinspector -lSDWebImage -lSDWebImageWebPCoder -lc++ -lglog -llibwebp -lreact-native-google-maps -lreact-native-maps -lreact-native-splash-screen -lstdc++ -lyoga -lz -framework Accelerate -framework CoreData -framework CoreGraphics -framework CoreImage -framework CoreLocation -framework CoreTelephony -framework CoreText -framework GLKit -framework GoogleMaps -framework GoogleMapsBase -framework GoogleMapsCore -framework ImageIO -framework JavaScriptCore -framework OpenGLES -framework QuartzCore -framework SystemConfiguration -framework UIKit -ObjC -lc++ -Xlinker -sectcreate -Xlinker __TEXT -Xlinker __entitlements -Xlinker /Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Intermediates.noindex/Venn.build/Debug-iphonesimulator/Venn.build/Venn.app-Simulated.xcent -lPods-Venn -Xlinker -dependency_info -Xlinker /Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Intermediates.noindex/Venn.build/Debug-iphonesimulator/Venn.build/Objects-normal/x86_64/Venn_dependency_info.dat -o /Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/Venn.app/Venn
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_UXCam", referenced from:
      objc-class-ref in libRNUxcam.a(RNUxcam.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)


** BUILD FAILED **


The following build commands failed:
	Ld /Users/omridoron/dev/venn/mobile-app-v2/ios/build/Venn/Build/Products/Debug-iphonesimulator/Venn.app/Venn normal x86_64
(1 failure)
```

updated the podspec like this PR
ttps://github.com/negativetwelve/react-native-ux-cam/pull/85
build still failed

**Comment by Zbish** [comment]
"react": "16.8.3",
"react-native": "0.60.3",
"react-native-ux-cam": "^5.1.2",

**Comment by richardgroves** [comment]
Hi - support for react 0.60 is something we are looking to add in a future release.

There appear to be some non-trivial changes that need to be made to the package, and we need to consider the impact on versions <0.60 as well.

**Comment by balramkhichar** [comment]
@richardgroves Same issue with "react-native": "0.57.1"

**Comment by richardgroves** [comment]
Hi - I've just pushed a change to the RNUxcam.podspec file that includes a dependency on the UXCam CocoaPod - you can edit your local version of the file to add the change, and run `pod update` in the ios folder of your app.

Then `react-native run-ios` _should_ find the proper library and link your app.

The problem appears to be in the RNUxcam.podspec not also including the library file in it directions to the linker. The add of `s.dependency 'UXCam'` was the quickest way to work around before we can deduce if there is a better way, or if that actually is a good way - it will mean new installs of the react-native module automatically use the latests UXCam pod, so it might end up being best to version limit that dependency. We will investigate.


For react-native 0.60 the install integration now seem to be:

`yarn add react-native-ux-cam`
`react-native link react-native-ux-cam`
`cd ios`
`pod update`
`cd ..`

Then add the usual code to your App.js:

`import RNUxcam from 'react-native-ux-cam';`
`RNUxcam.optIntoSchematicRecordings();`
`RNUxcam.startWithKey('YOUR API KEY');`

And then:

`react-native run-ios`

to build and start your app.

**Comment by richardgroves** [comment]
@balramkhichar I'm not sure if the change above will fix 0.57.1

**Comment by richardgroves** [comment]
Hi Michael (@mtford90)

Sorry about that - we had a lot of problems being able to integrate the MiniZip library into a binary - this is one we missed.

I've put a patched binary into https://github.com/negativetwelve/react-native-ux-cam/tree/richard-WIP and we'll do a proper update and release of the package in the next couple of days.

In the meantime if you get the UXCam.a file that has changed in that branch and replace the same file in your projects `node_modules/react-native-ux-cam/ios` folder and clean and rebuild it should have removed those duplicate symbols.

Richard
